server/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java

```diff
@@ -113,11 +115,17 @@ public ImmutableList<String> getStoredFileKeys() {
         .filter(ApplicantQuestion::isAnswered)
         .filter(ApplicantQuestion::isFileUploadQuestion)
         .map(ApplicantQuestion::createFileUploadQuestion)
-        .map(FileUploadQuestion::getFileKeyValue)
-        .flatMap(Optional::stream)
+        .flatMap(ReadOnlyApplicantProgramServiceImpl::getFileKeyValues)
         .collect(ImmutableList.toImmutableList());
   }
 
+  private static Stream<String> getFileKeyValues(FileUploadQuestion question) {
+    HashSet<String> fileKeys = new HashSet<>();
+    question.getFileKeyValue().ifPresent(fileKeys::add);
+    question.getFileKeyListValue().ifPresent(fileKeys::addAll);
```

Comment:
(question) Do we care if both getFileKey methods happen to add the same value? One will get "lost". Will it be handled later by a migrations?

Reply:
In this context we don't really care, it's just important that the file is present, it doesn't  matter which one it comes from.

But yeah, this is still a good point. I think in this case the danger is more about the fileKeyValue NOT being in the fileKeyValueList (which would happen if the applicant removes that file from the fileKeyvalueList). In this context would result in an applicant having access to that file when they actually shouldn't. So I'll stop combining this and just plumb the flag down.

FWIW My plan as documented had been to do a dual-write to avoid this issue (so when the flag is on we update fileKey whenever we update fileKeyList, so fileKey always has a subset of fileKeyList), but Nick is seeming to favor just a straight migration, which means we need to check the flag everywhere.